src/domain/synthetics/markets/useMarketsInfoRequest.ts

```diff
@@ -61,26 +62,281 @@ export type MarketsInfoResult = {
 export function useMarketsInfoRequest(chainId: number): MarketsInfoResult {
   const { address: account } = useAccount();
   const { marketsData, marketsAddresses } = useMarkets(chainId);
   const { tokensData, pricesUpdatedAt } = useTokensDataRequest(chainId);
-  const dataStoreAddress = getContract(chainId, "DataStore");
 
-  const isDepencenciesLoading = !marketsAddresses || !tokensData;
+  const isDependenciesLoading = !marketsAddresses || !tokensData;
+
+  const marketsValues = useMarketsValuesRequest({
+    chainId,
+    account,
+    isDependenciesLoading,
+    marketsAddresses,
+    marketsData,
+    tokensData,
+  });
+
+  const marketsConfigs = useMarketsConfigsRequest({
+    chainId,
+    isDependenciesLoading,
+    marketsAddresses,
+    marketsData,
+  });
+
+  const mergedData = useMemo(() => {
+    if (!marketsValues.data || !marketsConfigs.data || !marketsAddresses) {
+      return undefined;
+    }
+
+    // Manual merging to avoid cloning tokens as they are sometimes compared by reference
+    const data: MarketsInfoData = {};
+    for (const marketAddress of marketsAddresses) {
+      const market = marketsData?.[marketAddress];
+      const marketValues = marketsValues.data[marketAddress];
+      const marketConfig = marketsConfigs.data[marketAddress];
+
+      if (!market || !marketValues || !marketConfig) {
+        continue;
+      }
+
+      const longToken = getByKey(tokensData!, market.longTokenAddress)!;
+      const shortToken = getByKey(tokensData!, market.shortTokenAddress)!;
+      const indexToken = getByKey(tokensData!, convertTokenAddress(chainId, market.indexTokenAddress, "native"))!;
+
+      const fullMarketInfo: MarketInfo = {
+        ...marketValues,
+        ...marketConfig,
+        ...market,
+        longToken,
+        shortToken,
+        indexToken,
+      };
+
+      data[marketAddress] = fullMarketInfo;
+    }
+
+    return data as MarketsInfoData;
+  }, [marketsValues.data, marketsConfigs.data, marketsAddresses, marketsData, tokensData, chainId]);
+
+  return {
+    marketsInfoData: isDependenciesLoading ? undefined : mergedData,
+    tokensData,
+    pricesUpdatedAt,
+  };
+}
+
+function useMarketsValuesRequest({
+  chainId,
+  account,
+  isDependenciesLoading,
+  marketsAddresses,
+  marketsData,
+  tokensData,
+}: {
+  chainId: number;
+  account: string | undefined;
+  isDependenciesLoading: boolean;
+  marketsAddresses: string[] | undefined;
+  marketsData: MarketsData | undefined;
+  tokensData: TokensData | undefined;
+}) {
+  const dataStoreAddress = getContract(chainId, "DataStore");
 
-  const { data } = useMulticall(chainId, "useMarketsInfo", {
-    inWorker: true,
-    key: !isDepencenciesLoading &&
-      marketsAddresses.length > 0 && [marketsAddresses.join("-"), dataStoreAddress, account, pricesUpdatedAt],
+  const marketsValuesQuery = useMulticall(chainId, "useMarketsValuesRequest", {
+    key: !isDependenciesLoading && marketsAddresses!.length > 0 && [marketsAddresses, account],
```

Comment:
not sure if it's a critical or not but let's do `marketsAddresses.join('-')` just in case

Reply:
for swr stableHash library is used under the hood. so i dont see the reason to keep serializing the values in keys. whats your concert?